### PR TITLE
py-keras: new version

### PR DESCRIPTION
**Pull request by robert-mijakovic**

### CODE REVIEW

**Review by adamjstewart — CHANGES_REQUESTED** [review]
Can you also update the dependencies? Python 2 is no longer supported, only 3.6+. Also the install now requires:
```python
      install_requires=['numpy>=1.9.1',                                                  
                        'scipy>=0.14',                                                   
                        'pyyaml',                                                        
                        'h5py'],   
```

**Comment by robert-mijakovic** [comment]
> Can you also update the dependencies? Python 2 is no longer supported, only 3.6+. Also the install now requires:
> 
> ```python
>       install_requires=['numpy>=1.9.1',                                                  
>                         'scipy>=0.14',                                                   
>                         'pyyaml',                                                        
>                         'h5py'],   
> ```

I thought that I committed something that doesn't work at all. However, I tested and it seems to work.

I will, of course, add additional dependencies and restrict the use of python 2 to only older releases.

**Comment by tldahlgren** [comment]
> > Can you also update the dependencies? Python 2 is no longer supported, only 3.6+. Also the install now requires:
> > ```python
> >       install_requires=['numpy>=1.9.1',                                                  
> >                         'scipy>=0.14',                                                   
> >                         'pyyaml',                                                        
> >                         'h5py'],   
> > ```
> 
> I thought that I committed something that doesn't work at all. However, I tested and it seems to work.
> 
> I will, of course, add additional dependencies and restrict the use of python 2 to only older releases.

Are these changes going to be made in this or a separate PR?

**Review by tldahlgren — DISMISSED** [review]
Confirmed the sha256.

**Comment by adamjstewart** [comment]
Dependencies should be updated in this PR.

**Comment by robert-mijakovic** [comment]
> Dependencies should be updated in this PR.

Done!

File: var/spack/repos/builtin/packages/py-keras/package.py [hunk 1]
```diff
@@ -39,6 +40,11 @@ class PyKeras(PythonPackage):
     version('1.1.1', sha256='be1b67f62e5119f6f24a239a865dc47e6d9aa93b97b506ba34cab7353dbc23b6')
     version('1.1.0', sha256='36d83b027ba9d2c9da8e1eefc28f600ca93dc03423e033b633cbac9061af8a5d')
 
+    depends_on('python@3.6:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-numpy@1.9.1:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-scipy@0.14:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-pyyaml', type=('build', 'run'), when='@2.4:')
+    depends_on('py-h5py', type=('build', 'run'), when='@2.4:')
     depends_on('py-keras-applications', type='run', when='@2.2:')
     depends_on('py-keras-preprocessing', type='run', when='@2.2:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keras/package.py]: Many of these dependencies are no longer needed in the latest version.

File: var/spack/repos/builtin/packages/py-keras/package.py [hunk 2]
```diff
@@ -39,6 +40,11 @@ class PyKeras(PythonPackage):
     version('1.1.1', sha256='be1b67f62e5119f6f24a239a865dc47e6d9aa93b97b506ba34cab7353dbc23b6')
     version('1.1.0', sha256='36d83b027ba9d2c9da8e1eefc28f600ca93dc03423e033b633cbac9061af8a5d')
 
+    depends_on('python@3.6:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-numpy@1.9.1:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-scipy@0.14:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-pyyaml', type=('build', 'run'), when='@2.4:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keras/package.py]: This is already present below, so this line is redundant.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-keras/package.py [hunk 3]
```diff
@@ -39,8 +40,12 @@ class PyKeras(PythonPackage):
...
-    depends_on('py-keras-applications', type='run', when='@2.2:')
-    depends_on('py-keras-preprocessing', type='run', when='@2.2:')
+    depends_on('python@3.6:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-numpy@1.9.1:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-scipy@0.14:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-h5py', type=('build', 'run'), when='@2.4:')
+    depends_on('py-keras-applications', type='run', when='@2.2')
+    depends_on('py-keras-preprocessing', type='run', when='@2.2')
     depends_on('py-setuptools', type='build')
     depends_on('py-theano', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keras/package.py]: theano and six are no longer required in the latest version

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-keras/package.py (hunk 3 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/py-keras/package.py]: Where did you get the install dependencies? I was searching online but couldn't find them.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-keras/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#python-dependencies

File: var/spack/repos/builtin/packages/py-keras/package.py [hunk 4]
```diff
@@ -39,9 +40,13 @@ class PyKeras(PythonPackage):
...
+    depends_on('py-scipy@0.14:', type=('build', 'run'), when='@2.4:')
+    depends_on('py-h5py', type=('build', 'run'), when='@2.4:')
+    depends_on('py-keras-applications', type='run', when='@2.2')
+    depends_on('py-keras-preprocessing', type='run', when='@2.2')
     depends_on('py-setuptools', type='build')
-    depends_on('py-theano', type=('build', 'run'))
+    depends_on('py-theano', type=('build', 'run'), when='@2.2')
     depends_on('py-pyyaml', type=('build', 'run'))
-    depends_on('py-six', type=('build', 'run'))
+    depends_on('py-six', type=('build', 'run'), when='@2.2')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-keras/package.py]: I believe theano and six should be `@:2.2`, not `@2.2`

File: var/spack/repos/builtin/packages/py-keras/package.py (hunk 3 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/py-keras/package.py]: Thanks.

File: var/spack/repos/builtin/packages/py-keras/package.py (hunk 4 above)
Comment (robert-mijakovic) [reply var/spack/repos/builtin/packages/py-keras/package.py]: I guess now it is resolved :)

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**